tools/detekt/src/main/kotlin/com/datadog/tools/detekt/rules/UnsafeThirdPartyFunctionCall.kt

```diff
@@ -0,0 +1,292 @@
+        if (bindingContext == BindingContext.EMPTY) {
+            return
+        }
+        val resolvedCall = expression.getResolvedCall(bindingContext) ?: return
+        val call = resolvedCall.call
+        val returnType = expression.getType(bindingContext) ?: return
+
+        val receiverType = call.explicitReceiver?.type() ?: call.dispatchReceiver?.type()
+
+        if (receiverType == null) {
+            val arguments = resolvedCall.valueArguments
+                .map { it.key.type.lowerIfFlexible().fullType() }
+            val fullType = imports[returnType.toString()] ?: returnType.toString()
+            checkConstructorCall(expression, "$fullType.constructor(${arguments.joinToString()})")
+        } else {
+            val receiverFullType = receiverType.fullType()
+            val arguments = resolvedCall.valueArguments
+                .map { it.key.type.lowerIfFlexible().fullType() }
+            val calleeExpression = call.calleeExpression?.node?.text
+            checkFunctionCall(
+                expression,
+                "$receiverFullType.$calleeExpression(${arguments.joinToString()})",
+                calleeExpression
+            )
+        }
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun checkConstructorCall(expression: KtCallExpression, call: String) {
+        if (internalPackagePrefix.isNotEmpty() && call.startsWith(internalPackagePrefix)) {
+            return
+        }
+
+        if (knownThrowingCallsMap.containsKey(call)) {
+            val knownThrowables = knownThrowingCallsMap[call] ?: emptyList()
+            checkCallThrowingExceptions(expression, call, knownThrowables)
+        } else if (treatUnknownConstructorAsThrowing) {
+            val message = "Calling $call could throw exceptions, but this constructor is unknown"
+            reportUnsafeCall(expression, message)
+        }
+    }
+
+    private fun checkFunctionCall(
+        expression: KtCallExpression,
+        call: String,
+        functionName: String?
+    ) {
+        if (internalPackagePrefix.isNotEmpty() && call.startsWith(internalPackagePrefix)) {
+            return
+        }
+
+        if (functionName in kotlinHelperMethods) {
+            return
+        }
+
+        if (knownThrowingCallsMap.containsKey(call)) {
+            val knownThrowables = knownThrowingCallsMap[call] ?: emptyList()
+            checkCallThrowingExceptions(expression, call, knownThrowables)
+        } else if (treatUnknownFunctionAsThrowing && !knownSafeCalls.contains(call)) {
+            val message = "Calling $call could throw exceptions, but this method is unknown"
+            reportUnsafeCall(expression, message)
+        }
+    }
+
+    private fun checkCallThrowingExceptions(
+        expression: KtCallExpression,
+        call: String,
+        exceptions: List<String>
+    ) {
+        val catchesAnyException = caughtExceptions.any { list ->
+            list.any { e -> e in topLevelExceptions }
```

Comment:
I am wondering what happens if I catch the specific exception (e.g. `JsonParseException`) instead of the top `JavaException` or `KotlinException` class ? What would happen then ?

Reply:
Nevermind, I got my answer below.